homeassistant/components/tessie/config_flow.py

```diff
@@ -0,0 +1,102 @@
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize."""
+        self._reauth_entry: ConfigEntry | None = None
+
+    async def async_step_user(
+        self, user_input: Mapping[str, Any] | None = None
+    ) -> FlowResult:
+        """Get configuration from the user."""
+        errors: dict[str, str] = {}
+        if user_input and CONF_ACCESS_TOKEN in user_input:
+            try:
+                await get_state_of_all_vehicles(
+                    session=async_get_clientsession(self.hass),
+                    api_key=user_input[CONF_ACCESS_TOKEN],
+                    only_active=True,
+                )
+            except ClientResponseError as e:
+                if e.status == HTTPStatus.UNAUTHORIZED:
+                    errors["base"] = "invalid_access_token"
+                else:
+                    errors["base"] = "unknown"
+            except ClientConnectionError:
+                errors["base"] = "cannot_connect"
+            else:
+                return self.async_create_entry(
+                    title="Tessie",
+                    data=user_input,
+                )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=TESSIE_SCHEMA,
+            errors=errors,
+        )
+
+    async def async_step_reauth(self, user_input: Mapping[str, Any]) -> FlowResult:
+        """Handle re-auth."""
+        self._reauth_entry = self.hass.config_entries.async_get_entry(
+            self.context["entry_id"]
+        )
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: Mapping[str, Any] | None = None
+    ) -> FlowResult:
+        """Get update API Key from the user."""
+        errors: dict[str, str] = {}
+        assert self._reauth_entry
+        if user_input and CONF_ACCESS_TOKEN in user_input:
+            try:
+                await get_state_of_all_vehicles(
+                    session=async_get_clientsession(self.hass),
+                    api_key=user_input[CONF_ACCESS_TOKEN],
+                )
+            except ClientResponseError as e:
+                if e.status == HTTPStatus.UNAUTHORIZED:
+                    errors["base"] = "invalid_access_token"
+                else:
+                    errors["base"] = "unknown"
+            except ClientConnectionError:
+                errors["base"] = "cannot_connect"
+            else:
+                self.hass.config_entries.async_update_entry(
+                    self._reauth_entry, data=user_input
+                )
+                self.hass.async_create_task(
+                    self.hass.config_entries.async_reload(self._reauth_entry.entry_id)
+                )
+                return self.async_abort(reason="reauth_successful")
+
+        return self.async_show_form(
+            step_id="reauth_confirm",
+            data_schema=TESSIE_SCHEMA,
+            errors=errors,
+        )
```

Comment:
Usually we ask reauth to be added in the 2nd or 3rd PR for new integrations as it increases the review scope and slows down the process

Reply:
Understood, want me to rip it out or leave it as is?

Reply:
I'm fine with it either way, but it will go faster if you take it out since it will be less to review.

Since I can frequently find small blocks of time to do reviews, and very I have few large time blocks, small PRs almost always go in much quicker.

Reply:
Reauth has been removed